x509util/extensions.go

```diff
@@ -738,14 +754,16 @@ func MarshalSubjectAlternativeName(sans ...SubjectAlternativeName) (pkix.Extensi
 //
 // TODO(mariano,unreality): X400Address, DirectoryName, and EDIPartyName types
 // are defined in RFC5280 but are currently unimplemented
```

Comment:
The ACME DA draft RFC mentions the `hardware-module` identifier and `HardwareModule` SAN extension. I think it's the `id-on-hardwareModuleName`, as that exists on the same level of the tree as `PermanentIdentifier`. We may need to verify this is the intended one. 

`id-on-SmtpUTF8Mailbox` is an internationalized email address and defined in https://www.rfc-editor.org/rfc/rfc8398.txt. 

`id-on-dnsSRV` is a SRVName as defined in https://www.rfc-editor.org/rfc/rfc4985.html.

Not sure if the latter two make sense to add now.

Reply:
AFAIK, the format of that SAN is not yet defined, but I added this https://github.com/smallstep/crypto/issues/55

Reply:
Agree, I'm not entirely sure about the `id-on-hardwareModuleName`at the moment. I've noted this as a question for Brandon.